src/client/v2/algod/models/types.ts

```diff
@@ -4631,11 +4738,16 @@ export class SimulateUnnamedResourcesAccessed extends BaseModel {
     super();
     this.accounts = accounts;
     this.appLocals = appLocals;
-    this.apps = apps;
+    this.apps =
+      typeof apps === 'undefined' ? undefined : apps.map(ensureBigInt);
```

Comment:
I'm repeating myself somewhat as I think I'm understanding the issue better.  I think we ought to coercing all of the "omitEmpty" fields, to their "empty" values. So lists become `[]`, integers become `0` or `0n`.  I've only been commenting on the places you've had to edit because of the `bignum` change, but this seems like something we ought to do for all of these types before handing them to the user. _They_ shouldn't have to understand omitEmpty.

Reply:
I agree with this a lot. I'll think more about how this can be addresses

Reply:
It feels like it can be another PR, but would be very nice to do for v3.